server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -104,6 +108,40 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     });
   }
 
+  // ManagedDataSource
+
+  @Provides @Singleton ManagedDataSource writableDataSource(Environment environment,
+      KeywhizConfig config) {
+    DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
+    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "postgres-writable");
+    environment.lifecycle().manage(dataSource);
+
+    return dataSource;
+  }
+
+
+  @Provides @Singleton @Readonly ManagedDataSource readonlyDataSource(Environment environment,
+      KeywhizConfig config) {
+    DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
+    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "postgres-readonly");
+
+    return dataSource;
+  }
+
+  // Jooq
+
+  @Provides @Singleton DSLContext jooqContext(ManagedDataSource dataSource) throws SQLException {
+    return DSL.using(dataSource.getConnection());
+  }
+
+  // Uncomment @Readonly once DBI is gone.
+  @Provides @Singleton @Readonly DSLContext readonlyJooqContext(ManagedDataSource dataSource)
```

Comment:
You'd need to annotate the ManagedDataSource for it to be readonly: `readonlyJooqContext(@Readonly ManagedDataSource dataSource)`.

What's the comment mean? There's isn't a commented `@Readonly`.